java/sage/plugin/CorePluginManager.java

```diff
@@ -1946,97 +1967,67 @@ else if ("FanArt".equals(packages[i].type))
           return "FAILED - Invalid Package Type " + packages[i].type;
         }
 
-        java.io.File currFile = (java.io.File) packageMap.get(packages[i]);
-        // We need to check for overwrite at each file if its OK or not; we also need to stage any failed extractions
-        // zip file source, extract and preserve directory structure, permissions and such
-        if (Sage.DBG) System.out.println("Beginning zip file extraction from cache file " + currFile + " to root dest dir of: " + destDir);
-        java.util.zip.ZipFile zippy = null;
-        java.io.InputStream inStream = null;
-        java.io.FileOutputStream outStream = null;
         byte[] buf = new byte[65536];
-        try
+
+        java.io.File currFile = (java.io.File) packageMap.get(packages[i]);
+        if ("JAR".equals(packages[i].type) && currFile.getName().endsWith("jar"))
```

Comment:
I'd go with 'endsWith(".jar")' just so you're sure it's a file extension...although that corner case should never arise.